receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +66,359 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+}
+
+// scrapeScalarMetrics retrieves all SNMP data from scalar OIDs and turns the returned scalar data
+// into metrics with optional enum attributes
+func (s *snmpScraper) scrapeScalarMetrics(
+	metricHelper *otelMetricHelper,
+	configHelper *configHelper,
+	scraperErrors *scrapererror.ScrapeErrors,
+) {
+	metricScalarOIDs := configHelper.getMetricScalarOIDs()
+
+	// If no scalar metric configs, nothing else to do
+	if len(metricScalarOIDs) == 0 {
+		return
+	}
+
+	// Retrieve all SNMP data from scalar metric OIDs
+	scalarData := s.client.GetScalarData(metricScalarOIDs, scraperErrors)
+
+	// Create general resource if we're about to make scalar metrics
+	resource := metricHelper.getResource(generalResourceKey)
+	if resource == nil {
+		metricHelper.createResource(generalResourceKey, map[string]string{})
+	}
+
+	// For each piece of SNMP data, attempt to create the necessary OTEL structures (resources/metrics/datapoints)
+	for _, data := range scalarData {
+		if err := s.scalarDataToMetric(data, metricHelper, configHelper); err != nil {
+			scraperErrors.AddPartial(1, fmt.Errorf(errMsgScalarOIDProcessing, data.oid, err))
+		}
+	}
+}
+
+// scalarDataToMetric will take one piece of SNMP scalar data and turn it into a datapoint for
+// either a new or existing metric with attributes based on the related configs
+func (s *snmpScraper) scalarDataToMetric(
+	data SNMPData,
+	metricHelper *otelMetricHelper,
+	configHelper *configHelper,
+) error {
+	// Return an error if this SNMP scalar data is not of a useable type
+	if data.valueType == notSupportedVal || data.valueType == stringVal {
+		return fmt.Errorf(errMsgBadValueType, data.oid)
+	}
+
+	// Retrieve the metric config for this SNMP data
+	metricName := configHelper.getMetricName(data.oid)
+	metricCfg := configHelper.getMetricConfig(metricName)
+
+	// Create a new metric if needed
+	metric := metricHelper.getMetric(generalResourceKey, metricName)
+	if metric == nil {
+		if _, err := metricHelper.createMetric(generalResourceKey, metricName, metricCfg); err != nil {
+			return err
+		}
+	}
+
+	// Keys will be determined from the related attribute config and enum values will come straight from
+	// the metric config's attribute values.
+	dataPointAttributes := getScalarDataPointAttributes(configHelper, data.oid)
+
+	// Add data point to metric
+	if _, err := metricHelper.addMetricDataPoint(generalResourceKey, metricName, metricCfg, data, dataPointAttributes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// scrapeIndexedMetrics retrieves all SNMP data from column OIDs and turns the returned indexed data
+// into metrics with optional attribute and/or resource attributes
+func (s *snmpScraper) scrapeIndexedMetrics(
+	metricHelper *otelMetricHelper,
+	configHelper *configHelper,
+	scraperErrors *scrapererror.ScrapeErrors,
+) {
+	// Retrieve column OID SNMP indexed data for attributes
+	indexedAttributeValues := s.scrapeIndexedAttributes(configHelper, scraperErrors)
```

Comment:
```suggestion
	indexedAttributeValues := s.scrapeIndexedAttributes(configHelper. getAttributeColumnOIDs(), scraperErrors)
```

Reply:
@djaglowski we will also have to pass in the custom error message here as well. this makes this generic function look a little ugly imo, but if you think it's necessary for the PR to go through i'll combine this all into 1 func.